### DISCUSSION TITLE
Docs: Discuss emoji to represent rules that require type information

### DISCUSSION THREAD

**Discussion by bmish** [typescript-eslint/typescript-eslint#6007] [Technical Discussions]
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

The @typescript-eslint/eslint-plugin [rules list](https://typescript-eslint.io/rules/) uses the thought bubble 💭 emoji to represent rules that require type information. And I searched the top 1,000 ESLint plugins and found that ~4 other plugins have adopted this same convention.

Because I built [eslint-doc-generator](https://github.com/bmish/eslint-doc-generator) (which generates the README rules table and rule doc headers for an ESLint plugin, is used in 15+ of the top ESLint plugins, and has a goal of standardizing documentation across the ecosystem), I wanted to discuss whether the 💭 emoji is the best emoji to use for this purpose (corresponding issue in my tool https://github.com/bmish/eslint-doc-generator/issues/231).

Pros of keeping 💭:
- Less churn
- Has been in use here for a while and a few obscure plugins use it too

Pros of switching to 🔌:
- Better represents that additional / type information needs to be "plugged" in to the rule
- The thought bubble 💭 could be considered to represent an "idea" or "thought", which is very close to the meaning of the lightbulb 💡 emoji. The lightbulb emoji is already widely used to represent rules that provide suggestions. The overlapping meaning of these emojis could cause confusion.

So I have a slight preference toward switching to 🔌. What do you folks think?

My goal is to keep [eslint-doc-generator](https://github.com/bmish/eslint-doc-generator) in sync with you on this emoji since @typescript-eslint/eslint-plugin "owns" the concept of rules with type information.

If we do decide on a new emoji, I can easily update it in [eslint-doc-generator](https://github.com/bmish/eslint-doc-generator) to automatically roll out the update to many of the top plugins.

### Affected URL(s)

https://typescript-eslint.io/rules/

**Comment by JoshuaKGoldberg** [comment]
I'm in favor! Your points make sense and I greatly appreciate + support standardizing across the ecosystem. 🔌 ❤️ 

Would love to hear from other maintainers if they have any opinion on this.

Adding the `RFC` label because we'll likely enable GitHub Discussions soon, and I'll move this over to a discussion then.

**Comment by Josh-Cena** [comment]
FWIW I have a slight preference for 💭, though I don't know if my preference would have gone the other way had we started with 🔌. Type information is not "plugged" into your code; it's a parallel layer over the AST information that you can query from. I like how the thought bubble conveys the idea that it's a "semantic thought" and also how it's an overarching architecture instead of a "plug-in".

Theoretically thought bubble could be confusable with the lightbulb, though we need more user survey on that point.

**Reply by bmish** [reply]
Re-posting my reply from Nov 15 in the thread for organization:

--- 

@Josh-Cena those are fair points.

I was more so thinking of type information being "plugged" into the *rule* in that it enables the rule to understand more about what's going on in the code being linted than would otherwise be possible with plain JavaScript static analysis.

Your reasoning about a thought bubble representing how types add additional meaning to the code makes sense too.

If we can see a case of similar strength for both, or no obvious winner, then it might not be worth making any change.

Regarding the potential confusion between the thought bubble and the light bulb, this is more just a hypothetical concern, realistically I don't think it's likely to be much of an issue.

If anyone else has a preference or other emoji suggestion, let us know.

Goal is to figure out what would be the best possible emoji to represent this, and whether it's worth making a change.

**Reply by bmish** [reply]
Given the lack of consensus, it seems like we can just close this discussion for now and leave the emoji as-is. If anything changes, or if others chime in with thoughts, we can reconsider in the future.

**Reply by JoshuaKGoldberg** [reply]
Oh actually thanks for reminding me - I forget where, but I have a recollection of 🔌 "plug" referring to plugin. Which makes sense!

**Reply by bmish** [reply]
Good point, we may want to reserve 🔌 in case we need to represent the concept of plugins in the future.

**Comment by JoshuaKGoldberg** [comment]
I wish we had more electricity themed emojies. ⚡ is nice, but often used for performance - which would be ironic here.

**Comment by bmish** [comment]
Another related question is what emoji to use to represent TypeScript itself (there may be some overlap between this and representing the concept of "requiring type information"). I've used ⌨️ to represent TypeScript since "typing on a keyboard" evokes TypeScript just based on the sound ([example](https://github.com/bmish/eslint-doc-generator/blob/56a014f7a1757c98b83bfb06b21b0e525545b9ec/lib/emojis.ts#L8) where I'm using this emoji to represent any ESLint config related to types/TypeScript in [eslint-doc-generator](https://github.com/bmish/eslint-doc-generator)).

**Reply by JoshuaKGoldberg** [reply]
🦾 maybe? Since it's a powerful augmentation?

**Comment by JoshuaKGoldberg** [comment]
Closing this out as it's been >10 months without activity. But someone yell at me if there are emoji standardizations we should take on!
